be/src/env/env_hdfs.cpp

```diff
@@ -42,21 +42,16 @@ HdfsRandomAccessFile::~HdfsRandomAccessFile() {
     PLOG_IF(ERROR, r != 0) << "close " << _file_name << " failed";
 }
 
-Status HdfsRandomAccessFile::read(uint64_t offset, Slice* res) const {
-    tSize n = static_cast<tSize>(std::min<uint64_t>(std::numeric_limits<tSize>::max(), res->size));
-    tSize r = hdfsPread(_fs, _file, offset, res->data, n);
+StatusOr<int64_t> HdfsRandomAccessFile::read_at(int64_t offset, void* data, int64_t size) const {
+    tSize r = hdfsPread(_fs, _file, offset, data, size);
     if (UNLIKELY(r == -1)) {
         return Status::IOError(fmt::format("fail to hdfsPread {}: {}", _file_name, get_hdfs_err_msg()));
     }
-    res->size = r;
-    return Status::OK();
+    return r;
 }
 
-Status HdfsRandomAccessFile::read_at(uint64_t offset, const Slice& res) const {
-    if (UNLIKELY(res.size > std::numeric_limits<tSize>::max())) {
```

Comment:
why remove this protection?

Reply:
I didn't notice that. It has been restored, PTAL.